chasm/src/main/java/org/quiltmc/chasm/internal/ChasmContext.java

```diff
@@ -26,104 +25,31 @@ public ChasmContext(Context parent, ListNode classes) {
         this.classes = classes;
     }
 
-    private ClassInfo getClassInfo(String className) {
-        for (Node node : classes.getEntries()) {
-            MapNode classNode = NodeUtils.asMap(node);
+    @Override
+    public ClassInfo getClassInfo(String className) {
+        for (Node classNode : classes.getEntries()) {
             String name = NodeUtils.getAsString(classNode, NodeConstants.NAME);
             if (!name.equals(className)) {
                 continue;
             }
 
             String superName = NodeUtils.getAsString(classNode, NodeConstants.SUPER);
             int access = NodeUtils.getAsInt(classNode, NodeConstants.ACCESS);
-            ListNode interfacesList = NodeUtils.getAsList(classNode, NodeConstants.INTERFACES);
-            List<String> interfaces;
-            if (interfacesList == null) {
-                interfaces = Collections.emptyList();
-            } else {
-                interfaces = interfacesList.getEntries().stream().map(NodeUtils::asString).collect(Collectors.toList());
-            }
+            ListNode interfaces = NodeUtils.getAsList(classNode, NodeConstants.INTERFACES);
+
             return new ClassInfo(
-                    superName == null ? OBJECT : superName,
-                    (access & Opcodes.ACC_INTERFACE) != 0,
-                    interfaces
+                    className,
+                    superName == null ? ClassInfo.OBJECT : superName,
```

Comment:
If `superName` is null, then the superclass is null, not Object. The superclass of Object isn't Object. Maybe the purpose of this code was for a failsafe in case a transformer generates a class without a superclass? In which case you need to check if `className` is Object and allow nulls through in that case.

Reply:
We probably should support JDK classes even though it's not our usecase for Quilt Loader